test/test_gem_compatibility.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+# JRuby does not support forking, and it doesn't seem worth the effort to make it work.
+return unless Process.respond_to?(:fork)
+
+require_relative 'helper'
+
+describe 'gem compatibility' do
+  %w[
+    resolv-replace
+  ].each do |gem_name|
+    it "passes smoke test with #{gem_name.inspect} gem required" do
+      memcached(:binary, rand(21_397..21_896)) do |_, port|
+        in_isolation(timeout: 10) do
+          before_client = Dalli::Client.new("127.0.0.1:#{port}")
+
+          assert_round_trip(before_client, "Failed to round-trip key before requiring #{gem_name.inspect}")
+
+          require gem_name
+
+          after_client = Dalli::Client.new("127.0.0.1:#{port}")
+
+          assert_round_trip(after_client, "Failed to round-trip key after requiring #{gem_name.inspect}")
+        end
+      end
+    end
+  end
+
+  private
+
+  def assert_round_trip(client, message)
+    expected = SecureRandom.hex(4)
+    key = "round-trip-#{expected}"
+    ttl = 10 # seconds
+
+    client.set(key, expected, ttl)
+
+    assert_equal(expected, client.get(key), message)
+  end
+
+  def in_isolation(timeout:) # rubocop:disable Metrics
```

Comment:
This method triggers several offences in the `Metrics` department, but I don't know that splitting it would be an improvement.